switchbot/__init__.py

```diff
@@ -257,14 +262,15 @@ def _commandkey(self, key) -> str:
             key_suffix = ON_KEY_SUFFIX
         elif key == OFF_KEY:
             key_suffix = OFF_KEY_SUFFIX
+        elif key == DEVICE_BASIC_SETTINGS_KEY:
+            return KEY_PASSWORD_NOTIFY_PREFIX + self._password_encoded
         return KEY_PASSWORD_PREFIX + self._password_encoded + key_suffix
 
     def _writekey(self, key) -> bool:
         _LOGGER.debug("Prepare to send")
-        hand_service = self._device.getServiceByUUID(UUID)
-        hand = hand_service.getCharacteristics(HANDLE)[0]
+        hand = self._device.getCharacteristics(uuid=HANDLE)[0]
         _LOGGER.debug("Sending command, %s", key)
-        write_result = hand.write(binascii.a2b_hex(key), withResponse=True)
+        write_result = hand.write(binascii.a2b_hex(key), withResponse=False)
```

Comment:
Writing to characteristics does not return a response. (Verified in the wiki and nRFconnect android app)